src/Initializer/time_stepping/LtsWeights/LtsWeights.cpp

```diff
@@ -373,8 +410,246 @@ int LtsWeights::enforceMaximumDifferenceLocal(int maxDifference) {
     ++exchange;
   }
 
-#endif // USE_MPI
+#endif
 
   return numberOfReductions;
 }
+
+const LtsWeights::GlobalTimeStepDetails& LtsWeights::getDetails() const { return m_details; }
+
+const std::string& LtsWeights::getVelocityModel() const { return m_velocityModel; }
+
+unsigned LtsWeights::getRate() const { return m_rate; }
+
+std::vector<int>& LtsWeights::getVertexWeights() { return m_vertexWeights; }
+
+std::vector<double>& LtsWeights::getImbalances() { return m_imbalances; }
+
+const std::vector<int>& LtsWeights::getCellCosts() const { return m_cellCosts; }
+
+int LtsWeights::getVertexWeightElement() const { return m_vertexWeightElement; }
+
+int LtsWeights::getVertexWeightDynamicRupture() const { return m_vertexWeightDynamicRupture; }
+
+int LtsWeights::getVertexWeightFreeSurfaceWithGravity() const {
+  return m_vertexWeightFreeSurfaceWithGravity;
+}
+
+int LtsWeights::getNcon() const { return m_ncon; }
+
+const PUML::TETPUML* LtsWeights::getMesh() const { return m_mesh; }
+
+const std::vector<int>& LtsWeights::getClusterIds() const { return m_clusterIds; }
+
+LtsWeights::LtsWeights(const LtsWeightsConfig& config) noexcept
+    : m_velocityModel(config.velocityModel), m_rate(config.rate),
+      m_vertexWeightElement(config.vertexWeightElement),
+      m_vertexWeightDynamicRupture(config.vertexWeightDynamicRupture),
+      m_vertexWeightFreeSurfaceWithGravity(config.vertexWeightFreeSurfaceWithGravity),
+      m_nodeWeightModel(nullptr), m_edgeWeightModel(nullptr) {}
+
+void LtsWeights::setVertexWeights() { m_nodeWeightModel->setVertexWeights(); }
+
+void LtsWeights::setAllowedImbalances() { m_nodeWeightModel->setAllowedImbalances(); }
+
+int LtsWeights::evaluateNumberOfConstraints() const {
+  return m_nodeWeightModel->evaluateNumberOfConstraints();
+}
+
+void LtsWeights::setEdgeWeights(std::tuple<const std::vector<idx_t>&, const std::vector<idx_t>&,
+                                           const std::vector<idx_t>&>& graph) {
+  m_edgeWeightModel->setEdgeWeights(graph);
+}
+
+LtsWeights::~LtsWeights() noexcept {
+  delete m_nodeWeightModel;
+  delete m_edgeWeightModel;
+}
+
+void LtsWeights::addWeightModels(NodeWeightModel* nwm, EdgeWeightModel* ewm) {
+  m_nodeWeightModel = nwm;
+  m_edgeWeightModel = ewm;
+}
+
+std::vector<int>& LtsWeights::getEdgeWeights() { return m_edgeWeights; }
+
+int LtsWeights::find_rank(const std::vector<idx_t>& vrtxdist, idx_t elemId){
+  assert(!vrtxdist.empty());
+  assert(elemId < vrtxdist.back());
+  assert(elemId >= 0);
+  assert(vrtxdist[0] == 0);
+
+  for (size_t i = 0; i < vrtxdist.size()-1; i++){
+    if (elemId >= vrtxdist[i] && elemId < vrtxdist[i+1])
+    {
+      return i;
+    }
+  }
+
+  throw std::runtime_error("element id is not in valid range!");
+}
+
+
+std::vector<std::unordered_map<idx_t, int>>
+LtsWeights::exchangeGhostLayer(std::tuple<const std::vector<idx_t>&, const std::vector<idx_t>&,
```

Comment:
It is the distributed graph data is returned by METIS, the garph structure is then a tuple of 3: vertex distribution (over compute nodes), xadj (an array where xadj[i+1] and xadj[i] describe the offsets in the adjncy array in form of [begin,end) = [xadj[i], xadj[i+1]) , and adjncy is the vector that describes the end point of the edge. Since it is a tuple of 3 elements, I dont think I can give the references names, or can I?

Reply:
(there is actually 1 input, something was problematic with the formatting I guess)

Reply:
ha ok, now I understand, thank you.